### ISSUE TITLE
suggestion: tSNE map instead of PCA

### ISSUE THREAD

**Issue by SaraMati** [spyking-circus/spyking-circus#282]
can we have the option to choose the clustering algorithm to be applied on the tSNE space of the waveforms?

**Comment by yger** [comment]
SC does not use tSNE, mostly because I'm afraid that it might be very slow, and requires fine hand parameter tuning, depending on the dataset. But this is something we could think of: to launch tSNE before the clustering, instead of a plain and naive PCA. Do you have in mind a special library/implementation?

**Comment by SaraMati** [comment]
True, but the hyper parameters are a problem if the number of spikes is really low (~ 100) with anything on the scale of thousands, the performance is robust to tuning of the parameters (e.g. perplexity between 5 to 50). Anyways, a good choice can be perplexity of 30 and learning rate of 500. 

true regarding the speed. This implementation is fast though: https://github.com/DmitryUlyanov/Multicore-TSNE

it is nice if SC provides the users with the option to choose tSNE instead of PCA (through a setting in the parameter file perhaps), at least for experimentation. The default can still be PCA.
